frontend/src/pages/analytics/analytics-utils.ts

```diff
@@ -390,6 +390,47 @@ export const getPairEventAndConditions = (
           },
         },
       };
+      if (item.startEventRelationAttribute) {
+        pairEventAndCondition = {
+          ...pairEventAndCondition,
+          startEvent: {
+            ...pairEventAndCondition.startEvent,
+            retentionJoinColumn: {
+              category: defaultStr(
+                item.startEventRelationAttribute?.category,
+                ConditionCategory.OTHER
+              ),
+              property: defaultStr(item.startEventRelationAttribute?.name, ''),
+              dataType: defaultStr(
+                item.startEventRelationAttribute?.valueType,
+                MetadataValueType.STRING
+              ),
+            },
+          },
+        };
+      }
+      if (item.revisitEventRelationAttribute) {
+        pairEventAndCondition = {
+          ...pairEventAndCondition,
+          backEvent: {
+            ...pairEventAndCondition.backEvent,
+            retentionJoinColumn: {
+              category: defaultStr(
+                item.revisitEventRelationAttribute?.category,
+                ConditionCategory.OTHER
+              ),
+              property: defaultStr(
+                item.revisitEventRelationAttribute?.name,
+                ''
+              ),
+              dataType: defaultStr(
+                item.revisitEventRelationAttribute?.valueType,
+                MetadataValueType.STRING
+              ),
+            },
+          },
+        };
+      }
```

Comment:
It is better to check that start and revisit  retention join column has same datatype,  and throw an error if not equal.

Reply:
Added, it will validate data type of retention join column at frontend & backend.